payment/controllers.go

```diff
@@ -704,3 +705,102 @@ func VerifyCredential(service *Service) handlers.AppHandler {
 		return handlers.WrapError(nil, "Unknown credential type", http.StatusBadRequest)
 	})
 }
+
+// WebhookRouter - handles calls from various payment method webhooks informing payments of completion
+func WebhookRouter(service *Service) chi.Router {
+	r := chi.NewRouter()
+	r.Method("POST", "/stripe", middleware.InstrumentHandler("HandleStripeWebhook", HandleStripeWebhook(service)))
+	return r
+}
+
+// HandleStripeWebhook is the handler for stripe checkout session webhooks
+func HandleStripeWebhook(service *Service) handlers.AppHandler {
+	return handlers.AppHandler(func(w http.ResponseWriter, r *http.Request) *handlers.AppError {
+		ctx := r.Context()
+		// get logger
+		sublogger := logger(ctx).With().
+			Str("func", "HandleStripeWebhook").
+			Logger()
+
+		// get webhook secret from ctx
+		endpointSecret, err := appctx.GetStringFromContext(ctx, appctx.StripeWebhookSecretCTXKey)
+		if err != nil {
+			sublogger.Error().Err(err).Msg("failed to get stripe_webhook_secret from context")
+			return handlers.WrapError(
+				err, "error getting stripe_webhook_secret from context",
+				http.StatusInternalServerError)
+		}
+
+		b, err := requestutils.ReadWithLimit(r.Body, 65536)
+		if err != nil {
+			return handlers.WrapError(err, "error reading request body", http.StatusServiceUnavailable)
+		}
+
+		event, err := webhook.ConstructEvent(
+			b, r.Header.Get("Stripe-Signature"), endpointSecret)
+		if err != nil {
+			return handlers.WrapError(err, "error verifying webhook signature", http.StatusBadRequest)
+		}
+
+		// Handle invoice events
+		if event.Type == StripePaymentUpdated {
```

Comment:
Hm non-blocker, but I couldn't find any enums that we could reference in `stripe-go`. My gut reaction is that ideally the naming convention should best match https://stripe.com/docs/api/events/types like change from `StripePaymentUpdated` to `StripeInvoiceUpdated`. Thoughts?

Reply:
:+1:

Reply:
resolved: https://github.com/brave-intl/bat-go/pull/816/commits/6276de6cc2ffc85e725c37c13ded51bd438902a1